src/PaxfulApi.ts

```diff
@@ -68,9 +62,112 @@ export class PaxfulApi {
      * @param url - Url that should be called at api.paxful.com
      * @param payload - (Optional) Payload of the request
      */
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
-    public invoke(url: string, payload?: InvokeBody): Promise<any> {
-        return invoke(url, this.apiConfiguration, this.credentialStorage, payload);
+    public invoke(url: string, payload?: Record<string, unknown> | []): Promise<any> {
+        return invoke((
+            new RequestBuilder(`${process.env.PAXFUL_DATA_HOST}${url}`)
+                .acceptJson()
+                .withMethod("POST")
+                .withFormData(payload)
+        ), this.apiConfiguration, this.credentialStorage);
+    }
+
+    /**
+     * Uploads files on behalf of currently authenticated user,
+     * assumes that payload contains multipart form data.
+     *
+     * @param url - Url that should be called at api.paxful.com
+     * @param payload - (Optional) Payload of the request
+     * @param method - (Optional) Method to use. Default: POST
+     */
+    public upload(url: string, payload: Record<string, unknown> | [], method="POST"): Promise<any> {
+        return invoke((
+            new RequestBuilder(`${process.env.PAXFUL_DATA_HOST}${url}`)
+                .acceptJson()
+                .withMethod(method)
+                .withMultipartFormData(payload)
+        ), this.apiConfiguration, this.credentialStorage);
+    }
+
+    /**
+     * Downloads file on behalf of currently authenticated user,
+     * assumes that response will be binary file.
+     *
+     * @param url - Url that should be called at api.paxful.com
+     * @param method - (Optional) Method to use. Default: POST
+     */
+    public download(url: string, method="GET"): Promise<any> {
```

Comment:
Can we properly type hint a returned promise so it would have Promise<Buffer> type or something like that?

The same comment goes, in fact, to all other method in this class.

Reply:
Yes, we can use type alias.

Reply:
got it, will add